### ISSUE TITLE
Feature Request: Floorplans and visual inventory

### ISSUE THREAD

**Issue by jghanlon** [snipe/snipe-it#501]
Floorplans!  A visual inventory and overview of each floor.

This section would allow the import of an existing floorplan image to build on or the ability to design from scratch.  
Objects on the floorplan would tie in to other inventory databases so they are dynamically updated.
Multiple design layers could be overlayed to show all or some of the data.  A checkbox next to each layer controls visibility of the layer.

Layer categories could include icons or info boxes for:
- Users
- Phone extension
- Computer Asset numbers
- Computer Model
- Computer OS
- Security Cameras
- Wifi Access Points
- Server Rooms
- vlan/subnet
- Department
- Printers
- Room number

Floorplans could be printed out.

Edit mode and View mode.
Under View mode, you can select an object and the information about that object can be seen in an info box.  Example: select a workstation and see the specs, associated software licenses, associated user(s), history of repairs, history of moves, etc.

I currently build my floorplans using the original CAD drawings imported into Adobe Illustrator.  Having a dynamic and interactive floorplan inventory would be extremely useful and take much less time to keep up to date.

Thanks,
Jeff
##

**Comment by jghanlon** [comment]
Some sites for reference:
https://planstery.com/
http://www.imapbuilder.net/interactive-map-samples/flash-html5-interactive-floor-plan-map.php
http://stackoverflow.com/questions/6832964/interactive-floor-plans-in-html5
http://createhtml5map.com/interactive-map-blog/tag/floor-plan/

**Comment by jghanlon** [comment]
A few more possible tools:
http://openfloorplan.org/
https://inkscape.org/en/
http://www.fldraw.com/

**Comment by goetzk** [comment]
While we would find this useful too, I feel its a staggeringly complicated thing to build in to snipe. Given the time this would take I would rather other issues fixed and the option to upload images, cad files or pdfs to Locations (as can be done with images in models and assets) which could then be used for storing the floorpan for updating at a later date.

**Comment by jghanlon** [comment]
I understand the complexity issues.  I've mocked up something in Flash years ago but don't know anything about integrating with a database.  The PDF of the floor map would be a good start.  How difficult would it be to add a layer on top of that with some place-able objects that refer to the assets?  Could that be done in HTML5?

**Comment by lostdesign** [comment]
![Floorplan](https://i.imgur.com/E2rvhkn.png)

I added a floorplan by myself - used illustrator to draw it and then hooked it into the `dashboard.blade.php` + linked to profiles. I'll try to write something to do it within the tool.

**Comment by jarnokurlin** [comment]
I'm thinking about implementing this using [Mapael](https://www.vincentbroute.fr/mapael/). Mapael allows user to draw custom zoomable/pannable maps (also floor plans) using any SVG editor.

Plan:
-Allow user to add custom SVG maps (Add a new admin menu "Location maps") to system. Some geographical maps are already included with [Mapael maps](https://github.com/neveldo/mapael-maps).
-Add a location map setting to location objects -> binds locations to Mapael maps
-Add a new table "location_coordinates" which contains all the coordinates -> no need for new columns for assets/consumables/accessories/etc
-Add a new menu "Location Maps" -> user selects location and the map shows all placed objects which can be filtered and searched by categories/types/names. Unplaced objects can be added to map by double clicking desired point on the map (or by dragging them to right places?). Location maps shows predefined information about the object when user clicks/hovers it (user can select desired columns from location map settings).

Basically this allows to place everything on the map which has a location field.

Other thoughts:
-Add a new asset/consumable/accessory/component/etc by double clicking a point on the map?
-Allow user to select a custom icon to an object/asset category [https://rawgit.com/neveldo/jQuery-Mapael/2.1.0/examples/basic/plotted_cities_areas.html](https://rawgit.com/neveldo/jQuery-Mapael/2.1.0/examples/basic/plotted_cities_areas.html)

I was thinking to start this project after 4.0 version has been released. Any thoughts?

**Comment by hiboucochon** [comment]
This is a great feature. I've added 40 APs and would love to have a floor plan embedded to the dashboard

**Comment by Eelviny** [comment]
I'm the IT guy at a company that measures workplaces using sensors. We have two requirements:
1. Tracking of sensors, who is using what and where etc
2. Physical location of sensors within a floorplan, so we know which sensor corresponds to which desk

SnipeIT hits the first point precisely, but we can find no good solution out there which has both of these. I'm considering a proposal where I hire a dev to build this functionality into SnipeIT. I figured it would be good to put this into master, instead of just forking this project.

My question is: **If I build this, would the community/snipe be willing to implement & maintain it?** It would be a win-win for me and everyone else. No promises that this will go through - it depends on how open my company is feeling :wink:

**Comment by colttt** [comment]
3years later.. any news on it?

**Comment by snipe** [comment]
@colttt - if there was news on it, it would have been posted here. 

This is not currently on the short-term roadmap, but we would look at a PR if someone wants to submit one.

**Comment by colttt** [comment]
@Eelviny did you develop something?

**Comment by sactre** [comment]
+1
